libsolutil/CDCL.h

```diff
@@ -48,16 +49,79 @@ struct Literal
 		return std::make_tuple(positive, variable) < std::make_tuple(_other.positive, _other.variable);
 	}
 };
+
 using Clause = std::vector<Literal>;
 
+class TriState {
+public:
+	explicit constexpr TriState(const bool b) : val(b ? 1 : 0) {}
+	constexpr TriState() {}
+
+	bool operator==(const TriState& other) const
+	{
+		return val == other.val;
+	}
+
+	bool operator!=(const TriState& other) const
+	{
+		return val != other.val;
+	}
+
+	bool toBool() const {
+		if (val == 1)
+			return true;
+		else if (val == 0)
+			return false;
+		else
+		{
+			assert(val== 2);
+			assert(false && "UNSET cannot be converted to bool");
+		}
+	}
+
+	std::string toString()
+	{
+		if (val == 1)
+			return "true";
+		else if (val == 0)
+			return "false";
+		else
+		{
+			assert(val == 2);
+			return "unset";
+		}
+	}
+
+	constexpr static TriState t_true();
+	constexpr static TriState t_false();
+	constexpr static TriState t_unset();
+
+private:
+	// Default value is UNSET
+	uint8_t val = 2;
```

Comment:
```suggestion
	uint8_t m_value = 2;
```

Reply:
Ah, now I understand the rule! OK, got it! Quite a good rule actually, I should use it in my own code.